dev/com.ibm.ws.security.acme/src/com/ibm/ws/security/acme/internal/AcmeProviderImpl.java

```diff
@@ -873,5 +911,98 @@ protected void updateAcmeConfigService(AcmeConfigService acmeConfigService, Map<
 	public void setAcmeApplicationStateListener(AcmeApplicationStateListener acmeApplicationStateListener) {
 		applicationStateListenerRef.set(acmeApplicationStateListener);
 	}
+	
+	/**
+	 * This method determines if the directoryURI has changed by checking
+	 * the acme file.
+
+	 * @return True if the directoryURI has changed
+	 * @throws AcmeCaException
+	 */
+	private boolean directoryURIChanged() throws AcmeCaException {
+		File file = wslocation.getServerWorkareaResource("acme/acme.txt").asFile();
+		String fileDirURI = acmeConfig.getDirectoryURI();
+		try {
+		    BufferedReader br = new BufferedReader(new FileReader(file));
+		    String line, lastLine = null; 
+			while ((line = br.readLine()) != null && !line.isEmpty()) {
+				lastLine=line;
+			}
+			br.close();
+			if (lastLine != null && !lastLine.isEmpty()) {
+				StringTokenizer tok = new StringTokenizer(lastLine);
+				if (tok.hasMoreTokens()) tok.nextToken();
+				if (tok.hasMoreTokens()) tok.nextToken();
+				if (tok.hasMoreTokens()) fileDirURI = tok.nextToken();
+			}
+		} catch (Exception e) {
+			Tr.error(tc, e.getMessage()); // AcmeCaExceptions are localized.
+		}
+		String directoryURI = acmeConfig.getDirectoryURI();
+		if (directoryURI.equals(fileDirURI)) {
+			return false;
+		}
+		return true;
+	}
+	
+	/**
+	 * This method will create the initial acme file in the servers/workarea
+	 * directory. It holds certificate and directoryURI information, eg.
+	 *  Date                Serial                   DirectoryURI                    Account URI
+     *  --------------------------------------------------------------------------------------------
+     *  20200509231118      6542743894787011570      https://localhost:33827/dir     https://localhost:33827/my-account/1
+	 */
+	private void createAcmeFile() {
+		File file = wslocation.getServerWorkareaResource("acme/acme.txt").asFile();
+		if (file.exists()) return;
+		file.getParentFile().mkdirs();
+		try {
+			file.createNewFile();
+			FileWriter fr;
+			fr = new FileWriter(file, true);
+			fr.write("Date" + "\t\t\t" + "Serial" + "\t\t" + "DirectoryURI" + "\t\t" + "Account URI" + "\n");
+			fr.write("-------------------------------------------------------------------------------------------------------------------------");
+	       	fr.close();
+		} catch (IOException e) {
+			Tr.error(tc, e.getMessage()); // AcmeCaExceptions are localized.
+		}
+	}
+
+	private void updateAcmeFile() throws AcmeCaException {
+		updateAcmeFile(null);
+	}
+
+	/**
+	 * Update the acme file with the original or updated certificate
+	 * and directoryURI information.
+	 * @param acmeCertificate The AcmeCertificate to use to populate
+	 * the acme file or null to use the default certificate
+	 * 
+	 * @throws AcmeCaException
+	 */
+	private void updateAcmeFile(AcmeCertificate acmeCertificate) throws AcmeCaException {
```

Comment:
I don't see any logic in here to limit the number of lines. I think we should probably limit it to 10 or so.